### PR TITLE
fix: internal/test account event filters should not be queried via join

### PR DESCRIPTION
**Pull request by pauldambra**
In some circumstances we're doing something like

```
select
session_id
sum(thing)
from session_replay_events s
JOIN (select some_info, $session_id from events where blah)
where some_info = blah
group by session_id
```

if there are 7 rows in the events query we get 7 results in the join and so `sum(thing)` is 7x greater than it should be

if this has to be a join, we need to be joining on something which will have 0 or 1 rows per session

----

Ah it was #24558 - dang... unintended unexpected consequence not specifically protected by the tests 🙈 

----

In #24558 we removed the code that pushed event property queries down into a subquery because we don't allow them... but we do allow them via the "internal and test account" filter

by having them in a join we would have `${number of events in the session} * click count` clicks listed for a recording. for sessions with not many events this wasn't wrong enough for folk to notice

but in https://posthoghelp.zendesk.com/agent/tickets/18059 (moved to PostHog: https://us.posthog.com/project/2/support/tickets/20117) some recordings were showing as having >5M clicks 🤦

### CODE REVIEW

File: posthog/session_recordings/queries/session_recording_list_from_filters.py [hunk 1]
```diff
@@ -99,81 +122,17 @@ def run(self) -> SessionRecordingQueryResult:
...
-            group_by=[ast.Field(chain=["session_id"])],
-            having=self._having_predicates(),
+        return parse_select(
+            self.BASE_QUERY,
+            {
+                "order_by": self._order_by_clause(),
+                "where_predicates": self._where_predicates(),
+                "having_predicates": self._having_predicates(),
+            },
         )
```
Comment (pauldambra) [inline posthog/session_recordings/queries/session_recording_list_from_filters.py:132]: i just can't wrap my head around parsing queries this way 🙈 

## with parsing

with-parsing

## without parsing

without-parsing

so my guess is that the parsing is fast enough that legibility is more important  :)

File: posthog/hogql/database/schema/test/__snapshots__/test_session_replay_events.ambr [hunk 2]
```diff
@@ -112,6 +112,42 @@
...
+# ---
+# name: TestFilterSessionReplaysByEvents.test_select_by_subquery_on_event_property_without_join
+  '''
+  SELECT DISTINCT session_replay_events.session_id AS session_id
+  FROM session_replay_events
+  WHERE and(equals(session_replay_events.team_id, 2), in(session_replay_events.session_id,
+                                                           (SELECT events.`$session_id` AS `$session_id`
+                                                            FROM events
+                                                            WHERE and(equals(events.team_id, 2), ifNull(like(replaceRegexpAll(nullIf(nullIf(JSONExtractRaw(events.properties, '$current_url'), ''), 'null'), '^"|"$', ''), '%example.com%'), 0)))))
+  ORDER BY session_replay_events.session_id ASC
```
Comment (pauldambra) [inline posthog/hogql/database/schema/test/__snapshots__/test_session_replay_events.ambr:141]: wanted to get this in to prove to myself that the hogql subquery isn't being hoisted to a join 😊

File: posthog/session_recordings/queries/session_recording_list_from_filters.py [hunk 3]
```diff
@@ -463,15 +429,17 @@ def _event_predicates(self):
...
+            ),
             where=self._where_predicates(),
             having=self._having_predicates(),
             group_by=[ast.Field(chain=["$session_id"])],
         )
 
     def get_query_for_session_id_matching(self) -> ast.SelectQuery | ast.SelectUnionQuery | None:
         use_poe = poe_is_active(self._team) and self.person_properties
-        if self._filter.entities or use_poe:
+        if self._filter.entities or self.event_properties or use_poe:
```
Comment (pauldambra) [inline posthog/session_recordings/queries/session_recording_list_from_filters.py:442]: this is the thing we need to make sure that we're not joining from replay events to events

TBH we should follow up and remove that auto join it's too dangerous

**Review by Gilbert09 — APPROVED** [review]

**Comment by daibhin** [comment]
The original mistake was on me - sorry 😞 Didn't understand how it was avoiding the join and looked safe to remove (especially given the tests passed). Thanks for taking it on Paul 🙏

**Pull request merged**